specifyweb/specify/models.py

```diff
@@ -1442,6 +1447,7 @@ class Collectionobject(models.Model):
     yesno4 = models.BooleanField(blank=True, null=True, unique=False, db_column='YesNo4', db_index=False)
     yesno5 = models.BooleanField(blank=True, null=True, unique=False, db_column='YesNo5', db_index=False)
     yesno6 = models.BooleanField(blank=True, null=True, unique=False, db_column='YesNo6', db_index=False)
+    hasreferencecatalognumber = models.BooleanField(blank=True, null=True, unique=False, db_column='HasReferenceCatalogNumber', default=False, db_index=False)
```

Comment:
What is the purpose of this field? 
From [an internal document](https://docs.google.com/document/d/1vtM1VfRJDK4N3RwjIc13hvi8jdzSnejyiX2Z4AkbM8Q/edit) it is implied that this field will be used to indicate the CollectionObject is associated to another CollectionObject, and will not have its own catalogNumber and instead reference another CollectionObject's catalogNumber in some way. 

What is the intended and/or envisioned use case here? 
What does this provide that a CollectionObjectGroup (COG) does not?